core/dbt/parser/unit_tests.py

```diff
@@ -149,23 +147,30 @@ def parse_unit_test_case(self, test_case: UnitTestDefinition):
                 "name": input_name,
                 "path": f"{input_name}.sql",
             }
+            resource_type = original_input_node.resource_type
 
-            if original_input_node.resource_type in (
+            if resource_type in (
                 NodeType.Model,
                 NodeType.Seed,
                 NodeType.Snapshot,
```

Comment:
NodeType.Seed and NodeType.Snapshot are covered by the `elif` statement on line 167 and can be removed. 

I believe sending snapshot/seed nodes down this codepath is what is causing the test failures since seeds and snapshots don't have an `.version` attribute: 

```sh
21:17:51  1 of 1 START unit_test customers::test_is_valid_email_address .................. [RUN]
21:17:51  Unhandled error while executing 
'SnapshotNode' object has no attribute 'version'
21:17:51  1 of 1 ERROR customers::test_is_valid_email_address ............................ [ERROR in 0.01s]
21:17:51  
21:17:51  Finished running 1 unit test in 0 hours 0 minutes and 0.08 seconds (0.08s).
...
FAILED tests/functional/unit_testing/test_ut_snapshot_dependency.py::TestUnitTestSnapshotDependency::test_snapshot_dependency
```

Failing tests: https://github.com/dbt-labs/dbt-core/actions/runs/11786025311/job/32828487620?pr=10889

Reply:
@MichelleArk, the unit/functional tests are all good now